### PR TITLE
fix: Allow access codes on hidden tickets only

### LINKED ISSUE #4971: Access code should only be linked to hidden tickets
**Describe the bug**

Right now we are allowing access code to be linked to any ticket.

**To Reproduce**
1. Create an access code linking it to a public ticket

**Expected behavior**

We should only allow creating access codes for hidden tickets.

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #4971 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Right now we are allowing access code to be linked to any ticket.

#### Changes proposed in this pull request:
allow creating access codes only for hidden tickets.
Also fixed HTTP 500 in access code creating when ticket quantity is none.

##### Conflict
conflict

##### Success
success

### CODE REVIEW

File: app/api/schema/access_codes.py [hunk 1]
```diff
@@ -55,17 +55,16 @@ def validate_order_quantity(self, data, original_data):
...
                     "min-quantity should be less than max-quantity"
-                )
+            )
 
-        if 'tickets_number' in data and 'max_quantity' in data:
-            if data['tickets_number'] < data['max_quantity']:
-                raise UnprocessableEntity({'pointer': '/data/attributes/tickets-number'},
-                                          "tickets-number should be greater than max-quantity")
+        if tickets_number and max_quantity and (tickets_number < max_quantity):
+            raise UnprocessableEntity({'pointer': '/data/attributes/tickets-number'},
```
Comment (schedutron) [inline app/api/schema/access_codes.py]: Can it be equal to `max_quantity`?

Comment (srv-twry) [reply app/api/schema/access_codes.py]: @schedutron Does it answer your queries ? If yes, please approve or suggest other changes if you consider fit.

File: app/api/access_codes.py [hunk 2]
```diff
@@ -30,11 +31,33 @@ def before_post(self, args, kwargs, data):
...
+        """
+        if data.get('tickets', None):
+            for ticket in data['tickets']:
+                # Ensuring that the ticket exists and is hidden.
+                try:
+                    ticket_object = self.session.query(Ticket).filter_by(id=int(ticket),
+                                                                         deleted_at=None).one()
+                    if not ticket_object.is_hidden:
+                        raise ConflictException({'pointer': '/data/relationships/tickets'},
+                                                "Ticket with id {} is public."
```
Comment (bhaveshAn) [inline app/api/access_codes.py]: `+` isn't required while concatenation ?

File: app/api/access_codes.py [hunk 3]
```diff
@@ -30,11 +31,33 @@ def before_post(self, args, kwargs, data):
...
+                                                " Access code cannot be applied to public tickets".format(str(ticket)))
```
Comment (maxlorenz) [inline app/api/access_codes.py]: `str` isn't need here, right?

**Pull request merged**